core-bundle/src/Cache/EntityTagger.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Cache;
+
+use Doctrine\ORM\EntityManagerInterface;
+use Doctrine\ORM\Mapping\ClassMetadata;
+use Doctrine\Persistence\Mapping\MappingException;
+
+class EntityTagger
```

Comment:
Why is this called `Tagger` but it does not tag? It returns tags but it does not tag. Why does it not tag right away? ;)

Reply:
Returning the tags instead of tagging right away could have the benefit that you could use its methods for clearing the cache based on entities too maybe?

Reply:
Correct, that's the idea. 

But I agree that the naming could be improved. Any suggestions? :slightly_smiling_face:

Reply:
`EntityCacheTags`?

Reply:
Sure, I think I would inject the response tagger here and have two methods. One for fetching the tags and one for tagging right away (forwarding the tags to the response tagger). That way the name is correct and both options are available. Also, you then don't have to inject both, the response tagger plus this new service if you want to tag right away :)

Reply:
Just thinking: do we also want this concept to work with models? :thinking: Then we should maybe get rid of the `Entity` in the name.

Reply:
> Just thinking: do we also want this concept to work with models? 🤔 Then we should maybe get rid of the `Entity` in the name.

Imho yes. But I think the naming is fine. I see "Entity" as a general term and not exclusive to just doctrine entities.